packages/tasit-sdk/src/helpers/DecentralandUtils.js

```diff
@@ -50,19 +71,78 @@ export default class DecentralandUtils {
     return openOrders.map(order => order.values);
   };
 
-  #getCreatedSellOrders = async fromBlock => {
-    return this.#getOrders(fromBlock, "OrderCreated");
+  getAssetsOf = async address => {
+    const [estateIds, parcelIds] = await Promise.all([
+      this.getEstateIdsOf(address),
+      this.getParcelIdsOf(address),
+    ]);
+
+    const estateAddress = this.#estate.address;
+    const landAddress = this.#land.address;
+
+    const estates = estateIds.map(id => ({
+      id,
+      nftAddress: estateAddress,
+    }));
+
+    const parcels = parcelIds.map(id => ({
+      id,
+      nftAddress: landAddress,
+    }));
+
+    const assets = [...estates, ...parcels];
+
+    return assets;
   };
 
-  #getCancelledSellOrders = async fromBlock => {
-    return this.#getOrders(fromBlock, "OrderCancelled");
+  getEstateIdsOf = async address => {
+    const ids = await this.#getAssetIdsOf(this.#estate, address);
+    return ids;
   };
 
-  #getExecutedSellOrders = async fromBlock => {
-    return this.#getOrders(fromBlock, "OrderSuccessful");
+  getParcelIdsOf = async address => {
+    const ids = await this.#getAssetIdsOf(this.#land, address);
+    return ids;
   };
 
-  #getOrders = async (fromBlock, eventName) => {
+  #getAssetIdsOf = async (contract, address) => {
+    const getAssetId = transfer => `${transfer.assetId}`;
+
+    const transferEvents = await this.#getTransfers(contract);
+
+    const transfers = transferEvents.map(event => event.values);
+
+    const received = transfers
+      .filter(transfer => transfer.to === address)
+      .map(getAssetId);
+
+    const sent = transfers
+      .filter(transfer => transfer.from === address)
+      .map(getAssetId);
+
+    const receivedIds = new Set(received);
```

Comment:
Can you explain how you're using `Set` here?

Reply:
To removing duplicated IDs if the same assets were sent and received several times.

Reply:
Makes sense - thanks. Resolving this!